webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/Score.scala

```diff
@@ -0,0 +1,251 @@
+import com.normation.rudder.domain.reports.CompliancePercent
+import com.normation.rudder.domain.reports.ComplianceSerializable
+import com.normation.rudder.score.ScoreValue.A
+import com.normation.rudder.score.ScoreValue.B
+import com.normation.rudder.score.ScoreValue.C
+import com.normation.rudder.score.ScoreValue.D
+import com.normation.rudder.score.ScoreValue.E
+import com.normation.zio._
+import zio.Ref
+import zio.ZIO
+import zio.json.ast.Json
+import zio.syntax._
+
+sealed trait ScoreValue {
+  def value: String
+}
+
+object ScoreValue {
+  case object A extends ScoreValue { val value = "A" }
+  case object B extends ScoreValue { val value = "B" }
+  case object C extends ScoreValue { val value = "C" }
+  case object D extends ScoreValue { val value = "D" }
+  case object E extends ScoreValue { val value = "E" }
+
+  val allValues: Set[ScoreValue] = ca.mrvisser.sealerate.values
+
+  def fromString(s: String) = allValues.find(_.value == s.toUpperCase()) match {
+    case None    => Left(s"${s} is not valid status value, accepted values are ${allValues.map(_.value).mkString(", ")}")
+    case Some(v) => Right(v)
+  }
+}
+
+trait Score[T] {
+  def name:    String
+  def value:   ScoreValue
+  def message: String
+  def details: T
+}
+
+case class NoDetailsScore(name: String, value: ScoreValue, message: String)
+case class JsonScore(value: ScoreValue, name: String, message: String, details: Json) extends Score[Json]
+case class ComplianceScore(value: ScoreValue, message: String, details: ComplianceSerializable)
+    extends Score[ComplianceSerializable] {
+  override val name = "compliance"
+}
+
+case class SystemUpdateStats(
+    nbPackages:  Int,
+    security:    Option[Int],
+    updates:     Option[Int],
+    defect:      Option[Int],
+    enhancement: Option[Int],
+    other:       Option[Int]
+)
+
+case class SystemUpdateScore(value: ScoreValue, message: String, details: SystemUpdateStats) extends Score[SystemUpdateStats] {
+  override val name = "system-updates"
+}
+case class GlobalScore(value: ScoreValue, message: String, details: List[NoDetailsScore])
+
+object GlobalScoreService {
+  def computeGlobalScore(oldScore: List[NoDetailsScore], scores: List[Score[_]]): GlobalScore = {
+
+    val correctScores = scores.foldRight(oldScore) {
+      case (newScore, acc) =>
+        NoDetailsScore(newScore.name, newScore.value, newScore.message) :: acc.filterNot(_.name == newScore.name)
+    }
+    import ScoreValue._
+    val score         = if (correctScores.exists(_.value == E)) { E }
+    else if (correctScores.exists(_.value == D)) { D }
+    else if (correctScores.exists(_.value == C)) {
+      C
+    } else if (correctScores.exists(_.value == B)) {
+      B
+    } else A
+    GlobalScore(score, s"There is at least a Score with ${score.value}", correctScores)
+  }
+}
+
+trait ScoreEvent
+
+case class InventoryScoreEvent(nodeId: NodeId, inventory: Inventory)                  extends ScoreEvent
```

Comment:
It should not be Inventory here, but `CoreNodeFact` or `NodeFact` if the data is not in the first. Inventory is now an internal datastructure that will be removed (hopefully) in a later release

Reply:
(it's in `NodeFact`)